### PR TITLE
[SPARK-9674] Re-enable ignored test in SQLQuerySuite

### PR DESCRIPTION
**Pull request by andrewor14**
The original code that this test tests is removed in https://github.com/apache/spark/commit/9270bd06fd0b16892e3f37213b5bc7813ea11fdd. It was ignored shortly before that so we never caught it. This patch re-enables the test and adds the code necessary to make it pass.

@JoshRosen @yhuai

### CODE REVIEW

**Comment by andrewor14** [comment]
retest this please

**Comment by yhuai** [comment]
`TungstenAggregationIterator` part looks good.

**Comment by yhuai** [comment]
LGTM

File: sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/TungstenAggregationIterator.scala [hunk 1]
```diff
@@ -397,14 +397,20 @@ class TungstenAggregationIterator(
   private[this] var mapIteratorHasNext: Boolean = false
 
   ///////////////////////////////////////////////////////////////////////////
-  // Part 4: The function used to switch this iterator from hash-based
-  // aggregation to sort-based aggregation.
+  // Part 3: Methods and fields used by sort-based aggregation.
```
Comment (andrewor14) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/TungstenAggregationIterator.scala]: oops this should be Part 4

Comment (yhuai) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/TungstenAggregationIterator.scala]: I will have a PR to change other parts of this class. I can update the comment.

**Comment by yhuai** [comment]
Merging to master and branch 1.5.

**Comment by andrewor14** [comment]
The right JIRA number is actually SPARK-9677 by the way!

**Pull request closed without merging**